### DISCUSSION TITLE
My closure operation is not working

### DISCUSSION THREAD

**Discussion by emailstorbala** [nushell/nushell#11931] [General]
I was working in a script migration (bash -> nu shell). When I tried to run the script, I am facing an error like the below:

![image](https://github.com/nushell/nushell/assets/12377900/ee1c41d5-0807-45ae-a139-2e137cb8f9fa)

```
# get the child pids of the process
def getcpids [pid: int] -> list<int> {
  echo $"pid->($pid)<-"
  let ps_list = (ps | where ppid == $pid)
  print $"ps_list->($ps_list)<-"
  let cpids = ($ps_list | each {|pid| $pid.pid})
  print $"cpids->($cpids)<-"

  #let cpids = (ps | where ppid == $pid | each { |x| $x.pid})
  return $cpids
}
```
My original closure is commented. I tried to split and investigate. When i get the filtered list (ps_list) above and try to get the pid for each record, I am facing the above error. Can anyone help me here?

**Comment by fdncred** [comment]
I have no issues with it. I also corrected some syntax.
![image](https://github.com/nushell/nushell/assets/343840/9a6c9561-bff7-46ba-8685-82e58eca360e)

**Reply by emailstorbala** [reply]
@fdncred Yes, when I try to run the code separately it works. But somehow in the program, it doesn't work. Unable to decode the error message.

One more information I found. In my program, the input variable $ppid is sometimes calculated like below.
`let ppid = (open /usr/cachelogic/var/testhost.pid)`

As you can infer, the variable is a string. The `ps` utility expects an `int` when doing the pipe command comparision. Due to the incompatible variable types, the comparision fails and provides an empty set when run separately. Anyhow I am unable to simulate the actual error when running separately.

**Reply by fdncred** [reply]
I believe opening files will always give a string. If you don't want that, then you should use `| into int` to change the data type.

**Reply by emailstorbala** [reply]
yeah @fdncred , I was able to fix that immediately. But I was expecting an error in that scenario (ps pipeline comparision).

I was totally clueless for sometime and then figured out what was going wrong. An error at that comparision would have helped.

**Reply by fdncred** [reply]
Feel free to submit a PR to help with this.
